parsing/location.ml

```diff
@@ -765,6 +770,53 @@ let batch_mode_printer : report_printer =
   { pp; pp_report_kind; pp_main_loc; pp_main_txt;
     pp_submsgs; pp_submsg; pp_submsg_loc; pp_submsg_txt }
 
+
+let json_mode_printer: json_report_printer = fun json report ->
+  let kind = match report.kind with
+    | Report_error -> `Assoc["kind", `String "error" ]
+    | Report_warning w ->
+        `Assoc ["kind", `String "warning"; "id", `String w]
+    | Report_warning_as_error w ->
+        `Assoc ["kind",`String "error_from_warning"; "id", `String w]
```

Comment:
The constructor uses `warning_as_error`, and the JSON output `error_from_warning`, maybe we could consider using a single name in both places? Which is the better one?

Reply:
Idle note: we would also have the choice to use the kind `"error"`, with some extra metadata `"original_kind": "warning"` or something.

Reply:
Keeping the `kind` simple and reporting `error` with some added metadata seems simpler indeed.